### ISSUE TITLE
🏗️ Tech Debt: Deprecated transformers imports + duplicated code across Collators

### ISSUE THREAD

**Issue by vietanhrs** [dinhanhx/VisualRoBERTa#6]
### 1. Deprecated transformers imports — `src/vision_language.py` lines 21-34
```python
from transformers.generation_logits_process import LogitsProcessorList
from transformers.generation_stopping_criteria import ...
from transformers.generation_utils import ...
from transformers.generation_beam_search import ...
from transformers.pytorch_utils import torch_int_div
```
These modules have been moved in transformers >= 4.x:
- `transformers.generation_utils` → `transformers.generation`
- `torch_int_div` was removed (use `//` operator)
- `generation_logits_process`, `generation_stopping_criteria`, `generation_beam_search` are now under `transformers.generation`

This code will break with modern transformers versions.

### 2. Duplicated Collator code — `src/data.py`
`PretrainCollator`, `VisualQuestionAnswerCollator`, and `ImageCaptioningCollator` share ~90% identical code:
- `tensorize_image_batch` is copy-pasted identically across all three
- `make_labels` is identical in `VisualQuestionAnswerCollator` and `ImageCaptioningCollator`
- The attention mask extension logic is the same

### 3. Duplicated Tools code — `src/utils.py`
`ImageCaptioningTools`, `VisualQuestionAnswerTools`, and `ImageTextMatchingTools` all have nearly identical `process_raw_data` methods.

## Suggested Fix
1. Update imports to use `transformers.generation` namespace
2. Create a base collator class with shared logic
3. Create a base tools class with shared `process_raw_data`
